pkg/mqtt/topic_manager.go

```diff
@@ -1,101 +0,0 @@
-package mqtt
-
-import (
-	"sync"
-)
-
-type OnSubscribeHandler func(topic string)
-type OnUnsubscribeHandler func(topic string)
-
-// topicManager keeps track of subscribed topics of the mqtt broker by subscribing to broker topic events.
-// This allows to get notified when a topic is subscribed or unsubscribed
-type topicManager struct {
-	subscribedTopics        map[string]int
-	subscribedTopicsLock    sync.RWMutex
-	subscribedTopicsDeleted int
-
-	cleanupThreshold int
-
-	onSubscribe   OnSubscribeHandler
-	onUnsubscribe OnUnsubscribeHandler
-}
-
-func (t *topicManager) Subscribe(topicName string) {
-	t.subscribedTopicsLock.Lock()
-	defer t.subscribedTopicsLock.Unlock()
-
-	count, has := t.subscribedTopics[topicName]
-	if has {
-		t.subscribedTopics[topicName] = count + 1
-	} else {
-		t.subscribedTopics[topicName] = 1
-	}
-
-	if t.onSubscribe != nil {
-		t.onSubscribe(topicName)
-	}
-}
-
-func (t *topicManager) Unsubscribe(topicName string) {
-	t.subscribedTopicsLock.Lock()
-	defer t.subscribedTopicsLock.Unlock()
-
-	count, has := t.subscribedTopics[topicName]
```

Comment:
Why did you remove that whole logic with the counters?
This is essential in my opinion.

Reply:
the subscription counters are implemented in server.go. the topic manager is replaced by the subscriber manager, this counter is not necessary for subscriber manager.